### PR TITLE
Calcite sql filter: not like

### PR DESCRIPTION
**Pull request by ZhaoqiWang0605**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: service/src/main/scala/com/yahoo/maha/service/calcite/MahaCalciteSqlParser.scala [hunk 1]
```diff
@@ -252,7 +252,10 @@ case class DefaultMahaCalciteSqlParser(mahaServiceConfig: MahaServiceConfig) ext
       case SqlKind.BETWEEN =>
         ArrayBuffer.empty += BetweenFilter(toLiteral(operands(0)), toLiteral(operands(1)), toLiteral(operands(2))).asInstanceOf[Filter]
       case SqlKind.LIKE =>
-        ArrayBuffer.empty += LikeFilter(toLiteral(operands(0)), toLiteral(operands(1))).asInstanceOf[Filter]
+        if (sqlBasicCall.getOperator.getName.contains("NOT"))
```
Comment (pranavbhole) [inline service/src/main/scala/com/yahoo/maha/service/calcite/MahaCalciteSqlParser.scala:255]: is there any cleaner way to do this in sql cell ?
NOT can be lowercase as well.

Comment (ZhaoqiWang0605) [reply service/src/main/scala/com/yahoo/maha/service/calcite/MahaCalciteSqlParser.scala:255]: I used to try to use the `negated` field in SqlOperator, but in SqlOperator class this field does not exist. And tried to search online as well, only find negate() method which is not fit into this case
Screen Shot 2021-07-20 at 6 21 51 PM

Comment (ZhaoqiWang0605) [reply service/src/main/scala/com/yahoo/maha/service/calcite/MahaCalciteSqlParser.scala]: just found out a clearer way to do that:
`sqlBasicCall.getOperator.asInstanceOf[SqlLikeOperator].isNegated`

**Review by pranavbhole — APPROVED** [review]
Amazing @ZhaoqiWang0605 . 
Thank you very much

**Pull request merged**